src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/ServerSentEventsMessageFormatter.cs

```diff
@@ -69,56 +59,131 @@ private static void WritePayload(ReadOnlyMemory<byte> payload, Stream output)
             // end up with an empty payload buffer, BUT we need to write it as an empty 'data:' line, so we need
             // to use a condition that ensure the only time we stop writing is when we write the slice after the final
             // newline.
-            var keepWriting = true;
```

Comment:
Can we uber simplify this code and since it's now the slow path, just do a very simple PositionOf and Write?